knative-operator/Dockerfile

```diff
@@ -1,18 +1,36 @@
-FROM registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.22-openshift-4.17 AS builder
+# DO NOT EDIT! Generated Dockerfile for knative-operator/cmd/knative-operator.
+ARG GO_BUILDER=registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.21-openshift-4.17
+ARG GO_RUNTIME=registry.access.redhat.com/ubi8/ubi-minimal
 
-ENV BASE=github.com/openshift-knative/serverless-operator
-WORKDIR ${GOPATH}/src/${BASE}
+FROM $GO_BUILDER as builder
 
 COPY . .
 
-ENV GOFLAGS="-mod=vendor"
-RUN go build -o /tmp/operator ${BASE}/knative-operator/cmd/manager
+ENV CGO_ENABLED=1
+ENV GOEXPERIMENT=strictfipsruntime
+
+RUN go build -tags strictfipsruntime -o /usr/bin/main ./knative-operator/cmd/knative-operator
+
+FROM $GO_RUNTIME
+
+ARG VERSION=
+
+# Install zoneinfo.
+RUN microdnf install tzdata
 
-FROM registry.access.redhat.com/ubi8-minimal:latest
 USER 65532
 
-COPY --from=builder /tmp/operator /ko-app/operator
-# install manifest[s]
+COPY --from=builder /usr/bin/main /usr/bin/main
+
 COPY knative-operator/deploy /deploy
 
-ENTRYPOINT ["/ko-app/operator"]
+LABEL \
+      com.redhat.component="openshift-serverless-1--knative-operator-rhel8-container" \
```

Comment:
The string is missing the project name, see `--` in the string. 
Normally, it is read from `project.imagePrefix` from project yaml (https://github.com/openshift-knative/eventing/blob/release-v1.15/openshift/project.yaml#L3) and removing the "knative-" prefix. So, you might need to put this in project yaml:
```
project:
  imagePrefix: serverless
```
The word "serverless" will be twice in some of the strings. But that's not too bad.

Reply:
The current product image has these labels:
```
LABEL \
      com.redhat.component="openshift-serverless-1-knative-rhel8-operator-container" \
      name="openshift-serverless-1/knative-rhel8-operator" \
      version="1.33.2" \
      summary="Red Hat OpenShift Serverless 1 Knative Operator" \
      maintainer="serverless-support@redhat.com" \
      description="Red Hat OpenShift Serverless 1 Knative Operator" \
      io.k8s.display-name="Red Hat OpenShift Serverless 1 Knative Operator"
```

Reply:
So, we probably don't have to add another "serverless" in there. But maybe we need to sanitize the string somehow.

Reply:
Something like this to sanitize the strings: https://github.com/openshift-knative/hack/pull/230